validation/validation.go

```diff
@@ -10,7 +10,7 @@ import (
 
 const validNamePattern = `^[a-zA-Z0-9-_ ]+$`
 const maxNameLength = 250
-const validSpiffeIdPattern = `^\^spiffe://[a-zA-Z0-9.-]+(/[a-zA-Z0-9._-]+)*$`
+const validSpiffeIdPattern = `^\^?spiffe:\/\/[a-zA-Z0-9.\-*()\[\]]+(\/[/a-zA-Z0-9._\-*()\[\]]+)*\$?$`
```

Comment:
This is a regex that matches a regex that matches a (subset of) spiffe id(s).

Since we are matching a regex, we'll need to match characters like (,),[,],\, etc.